lib/src/fade.dart

```diff
@@ -68,9 +73,12 @@ class FadeAnimatedTextKit extends AnimatedTextKit {
     bool repeatForever = false,
     bool displayFullTextOnTap = false,
     bool stopPauseOnTap = false,
-  }) : super(
+  })  : assert(fadeInEnd < fadeOutStart,
+            'The "fadeInEnd" argument has to be lesser than "fadeOutStart"'),
```

Comment:
Good, I like the assertion!  However, that assertion should be on `FadeAnimatedText`, not just `FadeAnimatedTextKit`.

Minor, but perhaps "_must be less than_" is more concise than "_has to be lesser than_".

Reply:
@awhitford about the assertion, the correct method would be to put assertion on both or only on `FadeAnimatedText`?

Reply:
`FadeAnimatedText` definitely should have it.
A second assertion for `FadeAnimatedTextKit` is redundant, so mentioning it again violates the [DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).

Reply:
@awhitford alright got it. I have added it only for `FadeAnimatedText`.